nwplus/nwhacks2019
# tag applicants cloud function

:tickets: **Ticket(s)**: Closes #232 

---

## :construction_worker: Changes

Created a function for bulk tagging applicants

Format of request body:

```
{
  "applicantType": 0 || 1 (0 for hacker, 1 for volunteer),
  "applicants": array of applicant ids,
  "tagsToAdd": array of tag ids,
  "tagsToRemove": array of tag ids,
}
```

Request header must contain `{ "Authorization": <token>" }`. The token is obtained by doing the following: 

`firebase.auth().currentUser.getIdToken(/* forceRefresh */ true).then(token => ... )`

## :thought_balloon: Notes

In the implementation, I don't check the integrity of the applicant using `validate` before writing back to the database. I also don't check the existence of the tag in the `tags` collection for any of the tags being added or removed. Let me know if this concerns you

## :flashlight: Testing Instructions

* Ensure that tags are added/removed for only the desired applicants
* Ensure that no other fields are modified other than tags

- [ ] Test adding and removing tags for one applicant
- [ ] Test adding and removing tags for multiple applicants
- [ ] Test adding and removing tags for hackers and volunteers
